.github/workflows/main.yml

```diff
@@ -123,98 +123,54 @@ jobs:
         set -e
         bash ./scripts/ci/deploy_tensorflow_blade.sh
   CUDA11-TORCH171:
-    if: github.repository == 'alibaba/BladeDISC'
-    # The type of runner that the job will run on
-    runs-on: [self-hosted, gpu-t4]
-    # Steps represent a sequence of tasks that will be executed as part of the job
-    steps:
-    # Runs a single command using the runners shell
-    - name: Checkout
-      uses: actions/checkout@v2.4.0
-    - name: Build Dev Docker
-      run: |
-        set -e
-        git submodule sync
-        git submodule update --depth=1 --init --recursive
-        docker pull bladedisc/bladedisc:latest-devel-cuda11.0
-        docker build --cache-from bladedisc/bladedisc:latest-devel-cuda11.0 -t disc-dev-cuda11.0 \
-          --build-arg BASEIMAGE=nvidia/cuda:11.0-cudnn8-devel-ubuntu18.04 \
-          --build-arg DEVICE=cu110 \
-          -f docker/dev/Dockerfile .
-    - name: Build and Test DISC
-      run: |
-        set -e
-        nvidia-docker run --rm -t --user $(id -u):$(id -g) \
-          -v $HOME/.cache:$HOME/.cache \
-          -v /etc/passwd:/etc/passwd:ro \
-          -v /etc/group:/etc/group:ro \
-          -v $PWD:/disc \
-          -e GITHUB_WORKFLOW=$GITHUB_WORKFLOW \
-          -w /disc \
-          disc-dev-cuda11.0 bash ./scripts/ci/test_pytorch_blade.sh
-    - name: Deploy PyTorch Blade
-      if: github.event.ref == 'refs/heads/main'
-      env:
-        ALIYUN_DOCKER_USERNAME: ${{ secrets.ALIYUN_DOCKER_USERNAME }}
-        ALIYUN_DOCKER_PASSWORD: ${{ secrets.ALIYUN_DOCKER_PASSWORD }}
-        DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
-        DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
-        GITHUB_PULL_REQUEST: ${{ github.event.number }}
-        LOCAL_DEV_DOCKER: disc-dev-cuda11.0
-        REMOTE_DEV_DOCKER: bladedisc:latest-devel-cuda11.0
-        REMOTE_RUNTIME_DOCKER: bladedisc:latest-runtime-torch1.7.1
-        RUNTIME_BASEIMAGE: nvidia/cuda:11.0-cudnn8-devel-ubuntu18.04
-        DOCKERFILE: docker/runtime/Dockerfile.pytorch
-      run: |
-        set -e
-        bash ./scripts/ci/deploy_pytorch_blade.sh
+    uses: ./.github/workflows/torch_cuda_reusable.yml
```

Comment:
How about makeing TF job reuse the CUDA workflow?

Reply:
I want to split it into another PR.